cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/PreferredLeaderElectionGoalTest.java

```diff
@@ -60,6 +61,30 @@ public void testOptimize() throws KafkaCruiseControlException {
     }
   }
 
+  @Test
+  public void testOptimizeWithDemotedBrokers() throws KafkaCruiseControlException {
+    ClusterModel clusterModel = createClusterModel();
+    clusterModel.setBrokerState(0, Broker.State.DEMOTED);
+
+    PreferredLeaderElectionGoal goal = new PreferredLeaderElectionGoal();
+    goal.optimize(clusterModel, Collections.emptySet(), Collections.emptySet());
+
+    for (String t : Arrays.asList(TOPIC0, TOPIC1, TOPIC2)) {
+      for (int p = 0; p < 3; p++) {
+        List<Replica> replicas = clusterModel.partition(new TopicPartition(t, p)).replicas();
+        for (int i = 0; i < 3; i++) {
+          Replica replica = replicas.get(i);
+          // only the first replica should be leader.
+          assertEquals(i == 0, replica.isLeader());
+          if (clusterModel.broker(0).replicas().contains(replica)) {
+            // The demoted replica should be in the last position.
+            assertEquals(replicas.size() - 1, i);
```

Comment:
Should we also check `assertEquals(false, replica.isLeader());`?

Reply:
The for loop has already checked all the replicas.